include/seqan3/core/type_traits/function.hpp

```diff
@@ -12,7 +12,50 @@
 
 #pragma once
 
-#include <seqan3/core/platform.hpp>
+#include <functional>
+
+#include <seqan3/core/type_list/traits.hpp>
+
+namespace seqan3
+{
+// ----------------------------------------------------------------------------
+// function_traits
+// ----------------------------------------------------------------------------
+
+//!\cond
+template <typename function_t>
+struct function_traits;
+//!\endcond
+
+/*!\brief A traits class to provide uniform interface to the properties of a std::function type.
```

Comment:
```suggestion
/*!\brief A traits class to provide a uniform interface to the properties of an `std::function` type.
```

Reply:
why `an`?

Reply:
Also note, that std::function is a link to the cppreference doc.